### PR TITLE
Modified and copied the parser into Expression.ts.

### PR DESCRIPTION
**Pull request by marcushanses**
Added parsers for Value and InfixOperator.

### CODE REVIEW

File: parse/Expression.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		algebra = false
+		comparator = false
+		while (!algebra && !comparator && source.peek(i)) {
+			algebra = Source.binaryOperator.some(c => (source as Source).peek(i)?.value == c)
+			comparator = Source.comparator.some(c => (source as Source).peek(i)?.value == c)
+			i++
+		}
+		result = parseNextExpression(result ?? Number.MAX_SAFE_INTEGER, source)
+		if (!result && algebra)
+			break
```
Comment (robinnedergard) [inline parse/Expression.ts:30]: While statement should be able to do this

File: parse/Expression.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		if (r) {
+			result = r
+			break
+		}
+	}
+
+	return result || left
+}
+const expressionParsers: [
+	(source: Source, previous: Expression | number | undefined) => Expression | undefined | false,
```
Comment (robinnedergard) [inline parse/Expression.ts:50]: Why false?

**Review by robinnedergard — CHANGES_REQUESTED** [review]

File: parse/Expression.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	if (typeof source == "string") {
+		handler = handler || new Error.ConsoleHandler()
+		const tokens = lexer.tokenize(source, handler).toArray()
+		source = new Source(tokens, handler)
+	}
+	handler = handler instanceof Expression ? handler : undefined
+	let result: Expression | undefined
+	let algebra = false
+	let comparator = false
+	while (+!result ^ +algebra) {
```
Comment (robinnedergard) [inline parse/Expression.ts:19]: `!result != algebra`

(or `!result != !!algebra` but algebra is already a boolean)

**Review by robinnedergard — CHANGES_REQUESTED** [review]

File: parse/Value.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import * as lexer from "../lexer"
+import { Value } from "../Value"
+import { addExpression, parseNextExpression } from "./Expression"
+
+addExpression((source, previous) => {
+	const fetchedArray: string[] = []
+	let fetched: lexer.Token[] | lexer.Token | undefined
+	while ((fetched = source.fetchIf("identifier", ".")))
+		fetchedArray.push(fetched[0].value)
+	let result: Expression | undefined | false
```
Comment (robinnedergard) [inline parse/Value.ts:11]: let result: Expression | undefined

**Review by robinnedergard — CHANGES_REQUESTED** [review]

**Review by robinnedergard — APPROVED** [review]

**Review by fredrikbryntesson — APPROVED** [review]

**Pull request merged**